base/test.jl

```diff
@@ -908,4 +908,11 @@ function detect_ambiguities(mods...; imported::Bool=false)
     collect(ambs)
 end
 
+immutable GenericString <: AbstractString
+    string::AbstractString
```

Comment:
Because packages are going to want to use this type as well. Otherwise how do you test that your interfaces work with string types that are not `String` once that's the only standard string type?